unity-game/Assets/Scripts/ElevatorScripts/ManualElevator.cs

```diff
@@ -0,0 +1,226 @@
+﻿using UnityEngine;
+using System.Collections;
+
+public class ManualElevator : MonoBehaviour {
+
+    public ElevatorButtonController theElevatorButtonController;
+
+    public bool notMoving;
+    public bool nearbyButton;
+    public bool reachedDestination;
+    public bool onStartPoint;
+    public bool onEndPoint;
+
+    public GameObject objectToMove;
+    public GameObject theRightWall;
+    public GameObject theLeftWall;
+    public GameObject theStartLeftBarrier;
+    public GameObject theStartRightBarrier;
+    public GameObject theEndLeftBarrier;
+    public GameObject theEndRightBarrier;
+    public GameObject theStartPoint;
+    public GameObject theEndPoint;
+
+    public Transform startPoint;
+    public Transform endPoint;
+
+    public float moveSpeed;
+    public float readyTimer;
+
+    private Vector3 currentTarget;
+
+
+    void Start()
+    {
+        currentTarget = endPoint.position;
+        readyTimer = .5f;
+        notMoving = true;
+
+        theLeftWall.SetActive(false);
+        theRightWall.SetActive(false);
+        theStartLeftBarrier.SetActive(false);
+        theStartRightBarrier.SetActive(false);
+        theEndLeftBarrier.SetActive(false);
+        theEndRightBarrier.SetActive(false);
+    }
+
+
+
+    void Update()
```

Comment:
a majority of the time, these platforms will not be doing anything. All this logic should probably be in a coroutine that is activated when a player interacts with the button.

Additionally, please break these out into separate functions and call those in the coroutine. Maybe something like `toggleMovement`, `updateMoveToTarget`, etc.